azext_iot/operations/hub.py

```diff
@@ -1172,14 +1193,14 @@ def iot_edge_deployment_list(
         login=login,
     )
 
-    filtered = [c for c in result if c["content"].get("modulesContent")]
+    filtered = [c for c in result if c["content"].get("modulesContent") is not None]
     return filtered[:top]
 
 
 def _iot_hub_configuration_list(
-    cmd, hub_name=None, top=10, resource_group_name=None, login=None
+    cmd, hub_name=None, top=None, resource_group_name=None, login=None
 ):
-    top = _process_top(top, upper_limit=100)
+    top = _process_top(top, upper_limit=CONFIG_LIST_UPPER_LIMIT)
```

Comment:
Looking at this now, I feel like we shouldn't bother setting a top upper limit. Instead set the default top to 1000. There is cap on configurations (but a soft cap), and would be good to not have to change this client side rule in the future.